### PR TITLE
MB-4403: adds gov accounts to allowed list

### PR DESCRIPTION
**Pull request by kilbergr**
## Description

This PR adds the gov accounts for MM app to the list of callable aws-vault profiles. It's one step towards making commands using aws-vault usable by other (gov) accounts.

## Reviewer Notes

I'm not sure this is necessary. It's part of an effort to make vault.go compatible with profiles (aka AWS accounts) besides transcom-ppp/transcom-com-legacy. I wonder, however, whether we won't end up removing the vault functions from scripts and commands that assume the legacy account.

## Code Review Verification Steps

* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-4403) for this change\

### CODE REVIEW

**Comment by carterjones** [comment]
I like what I'm seeing. I have to ask though, if you're not sure this is necessary, have you seen something break if you didn't already have this code in place? I'm wondering what makes you think it is necessary. I was very hesitant to ask because in my mind this makes sense, but I'm hesitant to add code if it may cause confusion down the road.

**Review by travelar — APPROVED** [review]

File: pkg/cli/vault.go [hunk 1]
```diff
@@ -34,6 +34,12 @@ const (
 	VaultAWSProfileTranscomPPP string = "transcom-ppp"
 	// VaultAWSProfileTranscomComLegacy is the aws-vault profile name for transcom-com-legacy
 	VaultAWSProfileTranscomComLegacy string = "transcom-com-legacy"
+	// VaultAWSProfileTranscomGovExp is the aws-vault profile name for transcom-gov-exp
+	VaultAWSProfileTranscomGovExp string = "transcom-gov-exp"
```
Comment (travelar) [inline pkg/cli/vault.go:38]: Should this be transcom-gov-milmove-exp? Same for below environments...

File: pkg/cli/vault.go [hunk 2]
```diff
@@ -34,6 +34,12 @@ const (
...
+	// VaultAWSProfileTranscomGovStg is the aws-vault profile name for transcom-gov-stg
+	VaultAWSProfileTranscomGovStg string = "transcom-gov-stg"
```
Comment (travelar) [inline pkg/cli/vault.go:40]: transcom-gov-milmove-stg?

File: pkg/cli/vault.go [hunk 3]
```diff
@@ -34,6 +34,12 @@ const (
...
+	// VaultAWSProfileTranscomGovPrd is the aws-vault profile name for transcom-gov-prd
+	VaultAWSProfileTranscomGovPrd string = "transcom-gov-prd"
```
Comment (travelar) [inline pkg/cli/vault.go:42]: transcom-gov-milmove-prd?

**Review by travelar — COMMENTED** [review]
I might have been a bit too quick on the approve button. I've got a few comments.

**Review by esacteksab — CHANGES_REQUESTED** [review]
What Erik said.

**Comment by kilbergr** [comment]
@carterjones nothing's breaking now because a lot of the cmds that use this aren't operating on govcloud. If you look at the ticket: https://dp3.atlassian.net/browse/MB-4403, it's asking to both add the new accounts to vault and to audit and create tickets for the work that ports cmds to govcloud. The idea is one day to remove the com accounts from vault completely. However, it seems to me that we might want to pursue removing vault from these commands instead, as we've done here: https://github.com/transcom/mymove/pull/5542/files#diff-2aa047f454f5bd799760283c7042c938d4ebe4697c997f36f5e392f0ddc13cc6L275 and https://github.com/transcom/mymove/pull/5394/files#diff-df0e528d0e9c41df8c421b2cab73303abd2f7861d9b405d683a89d1de484d0f2L237. I have a question for Paul in slack  about the intent: https://ustcdp3.slack.com/archives/CP496B8DB/p1609973738029500 to try to clarify.

**Review by esacteksab — APPROVED** [review]

**Comment by kilbergr** [comment]
Closing because we will be taking this library out of the repo.

**Pull request closed without merging**